fedbiomed/common/metrics.py

```diff
@@ -369,7 +369,7 @@ def _configure_y_true_pred_(y_true: Union[np.ndarray, list],
         if y_true.ndim == 0:
             y_true = y_true.reshape((1,))
 
-        if len(y_pred) != len(y_true):
+        if y_pred.shape[0] != y_true.shape[0]:
```

Comment:
Hi @ybouilla, 

I don't but what about removing `y_pred = np.squeeze(y_pred)` and `y_true = np.squeeze(y_true)`  to avoid getting error if batch size is equal to 1. Do you think it will create some regression bugs? Because I only tested using MNIST and it seems that it is working.

Reply:
I tested a regression model using scikit-learn model,  and it failed due to non-squeezed predictions list.

Reply:
After checking some implementation in batch iterator. I saw that `NPDataLoader` adds another dimension of target variable if it is 1D.
```python
        # If the researcher gave a 1-dimensional dataset, we expand it to 2 dimensions
        if dataset.ndim == 1:
            dataset = dataset[:, np.newaxis]

        # If the researcher gave a 1-dimensional target, we expand it to 2 dimensions
        if target.ndim == 1:
            target = target[:, np.newaxis]
```
Issue can be solved by disabling `newaxsis` for `target` variable. Instead, it can be implemented in `BaseSkLearnModel.train` function. 


```python 
    def train(
        self,
        inputs: np.ndarray,
        targets: np.ndarray,
        stdout: Optional[List[List[str]]] = None,
        **kwargs,
    ) -> None:
        """Run a training step, and record associated gradients.

        Args:
            inputs: inputs data.
            targets: targets, to be fit with inputs data.
            stdout: list of console outputs that have been collected
                during training, that contains losses values.
                Used to plot model losses. Defaults to None.

        Raises:
            FedbiomedModelError: if training has not been initialized.
        """
        if inputs.ndim == 1:
            dataset = inputs[:, np.newaxis]

        # If the researcher gave a 1-dimensional target, we expand it to 2 dimensions
        if targets.ndim == 1:
            targets = targets[:, np.newaxis]

        batch_size = inputs.shape[0]
        w_init = self.get_weights()
        w_updt = {key: np.zeros_like(val) for key, val in w_init.items()}
        # Iterate over the batch; accumulate sample-wise gradients (and loss).
        for idx in range(batch_size):
            # Compute updated weights based on the sample. Capture loss prints.
            with capture_stdout() as console:
                self.model.partial_fit(inputs[idx : idx + 1], targets[idx])
            if stdout is not None:
                stdout.append(console)
            # Accumulate updated weights (weights + sum of gradients).
            # Reset the model's weights and iteration counter.
            for key in self.param_list:
                w_updt[key] += getattr(self.model, key)
                setattr(self.model, key, w_init[key].copy())
            self.model.n_iter_ -= 1
        # Compute the batch-averaged, learning-rate-scaled gradients.
        # Note: w_init: {w_t}, w_updt: {w_t - eta_t * sum_{s=1}^B(grad_s)}
        #       hence eta_t * avg(grad_s) = w_init - (w_updt / B)

        self._gradients = {
            key: w_init[key] - (w_updt[key] / batch_size)
            for key in self.param_list
        }
```

Reply:
Thanks for this finding: let's resolve conversation and update the related issue #1013